### PR TITLE
En verification server: add canary e2e test job

### PR DESCRIPTION
**Pull request by chaodaiG**
/assign @sethvargo

### CODE REVIEW

File: prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml [hunk 1]
```diff
@@ -33,3 +33,32 @@ presubmits:
           requests:
             cpu: 7
             memory: '16Gi'
+  - name: pull-en-verification-server-release-e2e-canary
+    cluster: build-apollo-server
+    always_run: true
```
Comment (sethvargo) [inline prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml:38]: I thought we werent running this all the time and were going to do a fixed interval instead?

Comment (chaodaiG) [reply prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml:38]: Sorry forgot to mention, this is for faster turnaround on PRs. Once the PR confirmed working I'll set the periodic runs right away

Comment (sethvargo) [reply prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml:38]: Will this fail our existing PRs though? Is there a way to mark it as "allow failure"?

Comment (chaodaiG) [reply prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml]: Yeah good catch, changed it to not always run

Comment (chaodaiG) [reply prow/prowjobs/google/exposure-notifications-verification-server/en-verification-server.yaml]: Also set optional to true so it's not blocking

**Pull request merged**